### PR TITLE
2568 input font sizes widths and button sizes

### PR DESCRIPTION
**Pull request by bameyrick**
### What is the context of this PR?
* Create input/field width classes
* Ensure select is same height as text inputs
* Ensure buttons are same height as inputs for use in inline groups
* Change focus / checked styling to be driven by CSS rather than JS
* Make date input more mobile friendly

### How to review
* Check width classes make sense
* Ensure focused/checked styling is still applied correctly
* Review date input and ideally test with users

### Deploy preview
https://deploy-preview-204--sdc-global-design-patterns.netlify.com/

### Breaking changes
* Select placeholders (e.g. 'Please select') are now display none in option lists: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-1562adaa907de8c36fc8a24c4c800c20R45
* Mobile base font size now matches desktop: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-3314b0e9ef2da664ffc99f2b8aeb4910L8
* Input padding has been dropped from 0.6rem to 0.5rem https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-413036b74c400a478d1681913c21f0bfR2
* Input types (currency, unit, duration, etc.) no longer have specialised classes. Use just `input-type` and an `input-type--prefix` modifier for units such as currency: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-a4f0a22443971354cb8e63aed8d5ff5b
* Fieldgroups no longer have specialised classes (`fieldgroup--date`, `fieldgroup--duration`). They now share a generic `fieldgroup--inline` class: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-d71a01d2df8e281ad4281129ca8b81df
* Input focus/checked styles are now handled by CSS rather than JavaScript adding/removing classes: https://github.com/ONSdigital/sdc-global-design-patterns/pull/204/files#diff-2820a5e92afcecbe2f6697dee3cde559R100

### CODE REVIEW

File: components/02-form-elements/currency/currency.hbs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="input-type input-type--currency">
+<div class="input-type input-type--prefix">
+  <input class="input input-type__input" name="answer" id="answer" type="number" data-qa="input-text">
   <abbr title="{{label}}"  class="input-type__type" id="answer-type">{{currency}}</abbr>
```
Comment (jrbarnes9) [inline components/02-form-elements/currency/currency.hbs]: @bameyrick  Issue with the alignment of currency fields
screen shot 2018-11-12 at 16 01 51

Comment (bameyrick) [reply components/02-form-elements/currency/currency.hbs]: screenshot 2018-11-13 at 11 37 12

Comment (bameyrick) [reply components/02-form-elements/currency/currency.hbs]: @jrbarnes9 This should be fixed now

**Comment by armstrongb** [comment]
I think that we will also need to check that moving to right alignment for numeric fields isn't going to affect users (esp. A11y users e.g. zoomtext)

**Comment by boxadesign** [comment]
As discussed. The checkbox and radio type border is lost when focused. Also the date example is borked.

**Comment by bameyrick** [comment]
@armstrongb having the unit on the right could also be an issue in zoomtext so I don't know whether that's something to solve here.

@boxadesign Those issues should be resolved now.

**Comment by armstrongb** [comment]
> @armstrongb having the unit on the right could also be an issue in zoomtext so I don't know whether that's something to solve here.

@bameyrick 
It is more that the entry of data / caret might be out of view (the unit is less of an issue as the question & label etc. should guide the user to enter the data in the right format anyhow). It could just require a DAC / A11y review to check that it isn't introducing an issue.

**Comment by will-madebymoon** [comment]
@armstrongb @bameyrick I don't think we need to get DAC to review this in order to make it live. We know that zoom text increases screen magnification x60. We also know that zoom text users will move right if they see content that is continuing of the screen. The alignment would only happen on units which would limited to the amount of digits you could enter thus restricting the size of the box. 

When we have another tranche of work we can ask them to review it as it's an easy change to make if it presents a problem but for now I think we should go with it.

**Comment by boxadesign** [comment]
image

image

Labels have been set to bold and missing 'Or'

**Comment by boxadesign** [comment]
The top two `select` elements shouldn't be full width:

image

Live: https://sdc-global-design-patterns.netlify.com/components/detail/select.html

**Comment by bameyrick** [comment]
@boxadesign Both of those issues should be resolved now 🤙🏻

**Pull request merged**